bundles/azure/aso/psql/bundle/config/01-flexible-server.ytt.yml

```diff
@@ -0,0 +1,229 @@
+  owner:
+    name: #@ flexibleServerName
+  charset: #@ db.charset
+  collation: #@ db.collation
+#@ end
+
+#@ fwRuleCounter = 0
+#@ for val in data.values.firewall_rules:
+#@   startAddr = ip.parse_addr(val.startIp)
+#@   endAddr = ip.parse_addr(val.endIp)
+--- 
+apiVersion: dbforpostgresql.azure.com/v1beta20210601
+kind: FlexibleServersFirewallRule
+metadata:
+  name: #@ data.values.resource_name + "-" + str(fwRuleCounter)
+  namespace: #@ data.values.resource_namespace
+spec:
+  owner:
+    name: #@ flexibleServerName
+  startIpAddress: #@ startAddr.string()
+  endIpAddress: #@ endAddr.string()
+#@   fwRuleCounter += 1
+#@ end  
+---
+apiVersion: dbforpostgresql.azure.com/v1beta20210601
+kind: FlexibleServer
+metadata:
+  name: #@ flexibleServerName
+  namespace: #@ data.values.resource_namespace
+spec:
+  location: #@ data.values.location
+  azureName: #@ flexibleServerName
+  owner:
+    name: #@ resourceGroupK8SName
+  version: #@ data.values.version
+  sku:
+    name: #@ data.values.instance_type
+    tier: #@ data.values.instance_tier
+  administratorLogin: #@ data.values.administrator_name
+  administratorLoginPassword:
+    name: #@ dbSecretName
+    key: password
+  storage:
+    storageSizeGB: #@ data.values.instance_storage_size_gb
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: #@ serviceAccountName
+  namespace: #@ serviceAccountNamespace
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: #@ roleName
+  namespace: #@ data.values.resource_namespace
+rules:
+- apiGroups:
+  - ""
+  resources:
+  - secrets
+  verbs:
+  - get
+  - list
+  - watch
+  resourceNames:
+  - #@ data.values.resource_name
+- apiGroups:
+  - dbforpostgresql.azure.com
+  resources:
+  - flexibleservers
+  - flexibleserversdatabases
+  - flexibleserversfirewallrule
+  verbs:
+  - get
+  - list
+  - watch
+  resourceNames:
+  - #@ data.values.resource_name
+- apiGroups:
+  - resources.azure.com
+  resources:
+  - resourcegroups
+  verbs:
+  - get
+  - list
+  - watch
+  resourceNames:
+  - #@ data.values.resource_name
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: #@ roleName
+  namespace: #@ data.values.resource_namespace
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: #@ roleName
+subjects:
+- kind: ServiceAccount
+  name: #@ serviceAccountName
+  namespace: #@ serviceAccountNamespace
+---
+apiVersion: secretgen.carvel.dev/v1alpha1
+kind: SecretTemplate
+metadata:
+  name: #@ data.values.resource_name + "-bindable"
+  namespace: #@ data.values.resource_namespace
+spec:
+  serviceAccountName: #@ serviceAccountName
+  inputResources:
+  - name: server
+    ref:
+      apiVersion: dbforpostgresql.azure.com/v1alpha1api20210601
+      kind: FlexibleServer
+      name: #@ flexibleServerName
+  - name: db
+    ref:
+      apiVersion: dbforpostgresql.azure.com/v1alpha1api20210601
+      kind: FlexibleServersDatabase
+      name: #@ data.values.resource_name
```

Comment:
question: Do we have a database named like that when I configure the thing with the following data values?
```yaml
[ .. ]
databases:
- name: "a-different-name"
  charset: "UTF8"
  collation: "en_US.utf8"
[ .. ]
```

Reply:
The databases get created correctly. We use the `az` CLI to verify the details of the FlexibleServer, and it reflects what is there.

But as stated elsewhere, we did not update the secretTemplate, so it might be best to rollback on this and add it after it is merged to the public repo.